src/main/java/com/klarna/hiverunner/hql/HiveQueryLanguageStatementFactory.java

```diff
@@ -0,0 +1,57 @@
+package com.klarna.hiverunner.hql;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.klarna.hiverunner.CommandShellEmulation;
+import com.klarna.hiverunner.sql.StatementsSplitter;
+
+public class HiveQueryLanguageStatementFactory {
+
+	private final Charset charset;
+	private final CommandShellEmulation commandShellEmulation;
+
+	public HiveQueryLanguageStatementFactory(Charset charset, CommandShellEmulation commandShellEmulation) {
+		this.charset = charset;
+		this.commandShellEmulation = commandShellEmulation;
+	}
+
+	public List<HiveQueryLanguageStatement> newInstanceForScript(String script) {
+		List<HiveQueryLanguageStatement> hqlStatements = new ArrayList<>();
+		List<String> statements = StatementsSplitter.splitStatements(commandShellEmulation.transformScript(script));
+		for (String statement : statements) {
+			hqlStatements.addAll(newInstanceForStatement(statement));
+		}
+		return hqlStatements;
+	}
+
+	public List<HiveQueryLanguageStatement> newInstanceForStatement(String statement) {
+		List<HiveQueryLanguageStatement> hqlStatements = new ArrayList<>();
+		String trimmedStatement = statement.trim();
+		if (commandShellEmulation.isImportFileStatement(trimmedStatement)) {
```

Comment:
I still think this will become very bloated if more commands are implemented.
There should be a clean separation between implementations of the different shells and the specific commands. I rather we pass the collection of statements through the emulator and it manipulates it in whatever way needed to simulate the commands.